lib/rendition.js

```diff
@@ -279,6 +282,26 @@ class Rendition {
         return obj;
     }
 
+    embed() {
+        return Number.isInteger(this.instructions.embedBinaryLimit)
+            && this.instructions.embedBinaryLimit <= EMBED_LIMIT_MAX
+            && this.size() <= this.instructions.embedBinaryLimit;
+    }
+
+    async asDataUri() {
+        // should never happen as embedBinaryLimit is limited, but extra safeguard against reading large files
+        if (this.size() > DATA_URI_LIMIT) {
+            throw new GenericError(`Rendition too large for data uri ${this.name}`);
+        }
```

Comment:
is this because asDataUri() is always expected to be invoked after embed() was called ? 
Ex:
 `data: rendition.embed() ? (await rendition.asDataUri()) : undefined`

Reply:
Data uris have a size limit of their own.

Reply:
(And that limit varies. So here we want to stay under DATA_URI_LIMIT. Take a look at https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs, the problems section for instance)

Reply:
yes, in theory this case should never be reached since `asDataUri()` is only used if `this.size()` < EMBED_LIMIT_MAX (32kb).